src/libpgagroal/pool.c

```diff
@@ -845,11 +847,21 @@ pgagroal_flush_server(signed char server)
       }
    }
 
-   if (config->number_of_users > 0 && config->number_of_limits > 0)
+   if (pgagroal_get_primary(&primary))
+      pgagroal_log_debug("No primary defined");
+   else
    {
-      if (!fork())
+      prefill = server != (unsigned char)primary && primary != -1;
```

Comment:
This needs to go into the L856 if check - and therefore you can remove  the `prefill` variable